File: Previous Works/task 23_02_2025/Pyramid.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
public class Pyramid{

	public static void main (String [] args){
		Scanner input = new Scanner (System.in);

		System.out.println("Enter pyramid level: ");
		int level = input.nextInt();

		int counter = level * 7;
		for (int count = 1; count <= level; count++){

			System.out.printf("%" + counter + "s", " ");

			counter -= 7;

			for (int number : pyramidLevelOutput(count)){
				System.out.printf("%-7d", number);

			}

			System.out.println();
		}


	}

	public static ArrayList <Integer> pyramidLevelOutput (int level){

		ArrayList <Integer> presentLevel = new ArrayList <Integer> ();

		if (level == 1){
			return new ArrayList <Integer> (List.of(1));
		}

		else{
			ArrayList <Integer> previousLevel = pyramidLevelOutput(level - 1);

			for (int index = 0; index <= (2*level) -2; index++){
				if (index == 0 || index == (2*level) -2){
					presentLevel.add(1);
				}
				else{
					presentLevel.add(previousLevel.get(index-1) * 3);
				}

			}

		}

		return presentLevel;

	}

}

File: Previous Works/task 23_02_2025/pyramid.py

def pyramid(levels):

	present_level = []

	if (levels == 1):

		return [1]

	else:
		previous_level = pyramid(levels - 1)

		for index in range((2*levels)-1):

			if (index == 0 or index == ((2*levels)-2)):

				present_level[index] = 1

			else:
				present_level[index] = previous_level[index - 1] * 3

		return present_level

pyramid_levels = int(input("Enter number of levels in the pyramid: "))

counter = pyramid_levels * 7

for value in range (1, pyramid_levels +1):

	print(f"{" ":>{counter}}", end = "")
	counter -= 7

	pyramidNumber = pyramid(value)

	for number in pyramidNumber:

		print(f"{number:<7}, end = """)

	print()